src/components/ui/Panel.jsx

```diff
@@ -145,6 +145,11 @@ class Panel extends React.Component {
     }
 
     this.setState({ currentHeight, holding: true });
+
+    // Show top_bar shadows when holding
+    const topBar = document.querySelector('.top_bar');
```

Comment:
I think we can add topBar on `this` to manipulate it through the component

Reply:
By doing this top bar manipulation in `<Panel>`, you're breaking the separation of concerns and introduce a new coupling between separated parts. You also introduce a new risk of inconsistent state if you forget some transitions. For exemple, this code doesn't manage the case of dragging from `'maximized'` and releasing to `'maximized'`. The shadow is restored but not removed at the end.

To avoid such problems, as this new class depends entirely on the `size` state, you need to find where it's actually stored and managed.
As we needed to control the size of the panel from outside, [we recenlty introduced a change in that](https://github.com/QwantResearch/erdapfel/pull/749): `size` is now passed as a prop to this component and managed in the state of PanelManager. So this is where your new logic should be moved. 

And that's good news, as there are already a lot of direct manipulation of the top bar element there :) We're beginning to see a future separated component emerge, when we'll be able to render the top bar in React.

Reply:
I totally underastand the problem shown here, though I'm don't think I can reliably update the topBar shadows in PanelManager: this component doesn't know when the panel is maximized, nor when it is being dragged. Only Panel knows that. I'm not sure how to change my code to move the logic here. @bbecquet

Reply:
> PanelManager doesn't know when the panel is maximized

Yes it does. As I explained in the previous comment, `PanelManager` is actually where the current panel size is stored now, in the state, so it can be controlled by stuff outside of the `Panel` component itself.
You can find it here: https://github.com/QwantResearch/erdapfel/blob/master/src/panel/PanelManager.jsx#L32
And as any React component state, you can set it, read it, and react to changes on it by using `componentDidUpdate`.

> nor when it is being dragged

That point is true, only the `Panel` knows that. But actually here you don't need to know when it's dragged, only when it's set to maximized.

Reply:
@xem Ok, I think I see your point. It's probably better visually if we restore the shadow as soon as we are dragging, is it what you wanted?

Reply:
@xem What about using z-index to cover the top bar-shadow ? It would be so much easier.
This way, we would only need to handle panel's shadow visibility

Reply:
@bbecquet yep it's what I meant.
Thanks a lot for these tips, I'm on it!

@G-Ray indeed, a bigger z-index in the panel_container div can invert the order of the panel and the top bar. I'm digging this way too!

Reply:
It works perfectly! We now have a pure CSS fix for this ticket :D